app/controllers/registrations_controller.rb

```diff
@@ -463,7 +455,8 @@ def payment_completion
     # Some API gateways like Stripe provide the client_secret as a kind of "checksum" (or fraud protection)
     #   back to us upon redirect. Other providers (like PayPal…) unfortunately don't.
     #   So we only compare this secret value with our stored intent record if it's actually provided to us
-    if secret_check.present? && stored_intent.client_secret != secret_check
+    validation_token = params[:payment_intent_client_secret]
```

Comment:
This was being returned by the `find_payment` method for some reason - it doesn't need to be, the method was just reading the value from the `params`

Reply:
It does need to be handled the way that it was handled prior to your changes.
The name of the param that it's being transported in is determined *by Stripe*. Only the Stripe connector/CPI would know that it needs to look up this param. PayPal may have other params or no params at all, and YourFunnyPaymentGateway123 may also compute the param differently.

TL;DR Reading from this specific param key is Stripe-specific and does not generalize to other payment methods

Reply:
Ah of course I'm with you

Reply:
Upon further consideration, I think the fact that this variable only has to do with Stripe seems a better argument for not expecting it to be returned from `find_payment_intent...` - as it forces us to return this value from that method for _all_ integrations - even when it is irrelevant to them. 

It still seems better to me to handle this in the method body, probably with an explicitly check for the payment 

Additionally - I think we should explicitly check the payment provider. Otherwise, it seems to me that the secret check could be bypassed by simply not including it